pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroup.java

```diff
@@ -70,9 +70,10 @@
 
     // Default ctor: it is not expected that anything outside of this package will need to directly
     // construct a ResourceGroup (i.e., without going through ResourceGroupService).
-    protected ResourceGroup(ResourceGroupService rgs, ResourceGroupConfigInfo rgConfig) {
+    protected ResourceGroup(ResourceGroupService rgs, String name,
+                            org.apache.pulsar.common.policies.data.ResourceGroup rgConfig) {
         this.rgs = rgs;
```

Comment:
Nit/qs: would it make sense to have the resource-group name within the org.apache.pulsar.common.policies.data.ResourceGroup class?

Reply:
My understanding is this is the artifact of how the data is laid out in znode in zookeeper. The name of the resource-group is part of the path to znode. The data is the fields in the ResourceGroup datastructure. I think embedding the name in the data will likely be confusing and wasteful.

Reply:
@bharanic-dev this would also mean adding name to the ResourceGroup class right? that would not be inline with the API exposed to the user.

Reply:
OK.